### ISSUE TITLE
Explainer config files are silently ignored: getattr() on a dict always returns the default

### ISSUE THREAD

**Issue by jubaeralmahmud23-ship-it** [L2R-UET/CFExpRec#2]
Hi — thanks for releasing the full benchmark with checkpoints and preprocessed data. Being able to
run eleven explainers against six recommenders without retraining anything is genuinely useful, and
it's what made our work possible.

While building on it we hit something that I think affects your reported numbers, so I wanted to
flag it rather than just work around it locally.

## The problem

Three of the graph explainers read their hyperparameters like this:

```python
# src/exp_model/cf_gnnexplainer.py:19-21
self.beta       = getattr(config, "beta", 0.5)
self.lr         = getattr(config, "lr", 1)
self.num_epochs = getattr(config, "num_epochs", 50)
```

But `config` arrives as a **dict**, from `exp_model_training.py`:

```python
with open(exp_config_path, "r", encoding="utf-8-sig") as f:
    exp_model_config = json.load(f)          # -> dict
exp_model = EXP_MODEL_REGISTRY[args.exp_model](rec_model, device, args, config=exp_model_config)
```

`getattr` looks up **attributes**, not keys. A dict has no attribute named `lr`, so every call falls
through to the default and the JSON file is never read. No exception, no warning.

## Which explainers are affected

| explainer | reads config via | status |
|---|---|---|
| GREASE, CLEAR, C2Explainer | `config.get(...)` | ✅ works correctly |
| **CF2, CF-GNNExplainer, UNR-Explainer** | `getattr(config, ...)` | ❌ config ignored |

Both idioms are present in the same release, which is what made this hard to spot.

## Concrete effect

Comparing each config file against the code default it actually uses:

| dataset | explainer | param | config says | code uses | ratio |
|---|---|---|---|---|---|
| Amazon / ML1M / Yahoo | CF-GNNExplainer | `lr` | 10 | 1 | **10×** |
| Amazon / ML1M / Yahoo | CF2 | `lr` | 1.5 | 0.5 | **3×** |
| Amazon / ML1M / Yahoo | CF2 | `num_epochs` | 30 | 20 | 1.5× |

Nine settings in total across the three datasets. Verified by instantiating `CFGNNExplainer` with the
config dict and printing `exp.lr` — it prints `1` while `cfgnnexplainer_config.json` contains `10`.

## Suggested fix

Either normalise at the call site in `exp_model_training.py`:

```python
exp_model_config = argparse.Namespace(**json.load(f))
```

or switch the three explainers to `config.get(...)` so they match the other three. A one-line
assertion in `ExpBaseModel.__init__` would also catch it permanently:

```python
assert not isinstance(config, dict) or not config, \
    "config passed as dict but read with getattr(); settings would be silently ignored"
```

## Why I'm reporting rather than patching quietly

If the published results were produced through this path, they reflect the code defaults rather than
the committed configurations. That doesn't necessarily change your conclusions — but it does mean
the config files in the repo don't describe the runs, which matters for a reproducibility benchmark
specifically.

I'm preparing a short paper that uses your benchmark and mentions this. It makes no claim of
negligence, credits the release, and is happy to reflect any correction you'd like to make first. If
you'd prefer to fix it before that appears, I'm glad to open a PR — and if I've misread the call
path, please tell me and I'll retract it.

Reproduction script available on request.

**Comment by huynq2k4** [comment]
Thank you for your interest in our paper! Regarding the configuration issue, we have fixed it. Since the code was developed by multiple people, there might have been some minor bugs introduced during the merging process that we did not catch initially. We have now checked and fixed the issue. Thanks again for pointing it out!
